File: OpenGUI/Assets/Plugins/OpenGUI/Scripts/OGPage.js
#pragma strict

public class OGPage extends MonoBehaviour {
	public var pageName : String = "New Page";

	public function StartPage () {}
	public function UpdatePage () {}
	public function ExitPage () {}
	public function DrawPage () {}

	public function UpdateStyles () {
		for ( var w : OGWidget in this.transform.GetComponentsInChildren.<OGWidget>(true) ) {
			if ( !w.root ) {
				w.root = OGRoot.GetInstance();
			}

			if ( !w.styles ) {
				w.ApplyDefaultStyles ();
			}
			
			w.styles.Refresh ( w.root.skin );
		}
	}

	public function ResetStyles () {
		for ( var w : OGWidget in this.transform.GetComponentsInChildren.<OGWidget>(true) ) {
			w.ApplyDefaultStyles ();
		}	
	}

}

File: OpenGUI/Assets/Plugins/OpenGUI/Scripts/OGPage.cs
using UnityEngine;
using System.Collections;

public class OGPage : MonoBehaviour {
	public string pageName = "New Page";

	public void StartPage () {}
	public void UpdatePage () {}
	public void ExitPage () {}
	public void DrawPage () {}

	public void UpdateStyles () {
		foreach ( OGWidget w in this.transform.GetComponentsInChildren<OGWidget>(true) ) {
			if ( w.root == null ) {
				w.root = OGRoot.GetInstance();
			}

			if ( w.styles == null ) {
				w.ApplyDefaultStyles ();
			}
			
			w.styles.Refresh ( w.root.skin );
		}
	}

	public void ResetStyles () {
		foreach ( OGWidget w in this.transform.GetComponentsInChildren<OGWidget>(true) ) {
			w.ApplyDefaultStyles ();
		}	
	}

}
